File: problems/HackerRank/maxMin.ts
function maxMin1(k: number, arr: number[]): number {
  arr.sort((a, b) => a - b);
  let windowStart = 0;
  let result = Infinity;

  for (let windowEnd = k - 1; windowEnd < arr.length; windowEnd++) {
    const diff = arr[windowEnd] - arr[windowStart];
    result = Math.min(result, diff);
    windowStart++;
  }
  return result;
}

console.log(maxMin1(4, [1, 2, 3, 4, 10, 20, 30, 40, 100, 200]));

File: problems/HackerRank/week5/maxMin.py
def maxMin(k, arr):
    arr.sort()
    window_start = 0
    ans = float("inf")
    
    for window_end in range(1, len(arr)):    
        if window_end - window_start + 1 == k:
            ans = min(ans, arr[window_end] - arr[window_start])
            window_start += 1
    return ans